src/models/accounts.js

```diff
@@ -8,15 +8,93 @@ const db = require('./db/accounts')
 const hashPassword = require('five-bells-shared/utils/hashPassword')
 const NotFoundError = require('five-bells-shared/errors/not-found-error')
 const UnauthorizedError = require('five-bells-shared/errors/unauthorized-error')
+const InvalidBodyError = require('five-bells-shared/errors/invalid-body-error')
+const uri = require('../services/uriManager')
+const validator = require('../services/validator')
+
+function getPublicData (data) {
+  return {
+    id: uri.make('account', data.name.toLowerCase()),
+    name: data.name
+  }
+}
+
+function convertFromExternal (data) {
+  // ID is optional on the incoming side
+  data = _.cloneDeep(data)
+  if (data.id) {
+    data.name = uri.parse(data.id, 'account').name.toLowerCase()
+    delete data.id
+  } else {
+    data.name = data.name.toLowerCase()
+  }
+
+  if (data.balance) {
+    data.balance = Number(data.balance)
+  }
+
+  // Passing in a password hash is a potential DoS vector because the hash
+  // specifies the number of iterations needed to verify it. So a malicious
+  // client could set it to UINT32_MAX and make the server do an insane amount
+  // of hashing work.
+  //
+  // There are other places in the code that should prevent users from setting
+  // the hash directly, but it's a good idea to put an extra layer of
+  // protection and prevent setting it here.
+  if (typeof data.password_hash !== 'undefined') {
+    delete data.password_hash
+  }
+
+  if (data.minimum_allowed_balance) {
+    if (data.minimum_allowed_balance === '-infinity') {
+      data.minimum_allowed_balance = Number.NEGATIVE_INFINITY
+    } else {
+      data.minimum_allowed_balance = Number(data.minimum_allowed_balance)
+    }
+  }
+
+  return data
+}
+
+function convertToExternal (data) {
+  data = _.cloneDeep(data)
+  data.id = uri.make('account', data.name.toLowerCase())
```

Comment:
I see that `name` is lowercased here, so I think you can disregard my comment in `src/controllers/accounts.js`.